bookkeeper-server/src/test/java/org/apache/bookkeeper/client/api/BookKeeperBuildersTest.java

```diff
@@ -0,0 +1,475 @@
+                .withWriteQuorumSize(1)
+                .withAckQuorumSize(0)
+                .withPassword(password)
+                .execute());
+            fail("shoud not be able to create a ledger with such specs");
+        } catch (BKIncorrectParameterException err) {
+        }
+
+        try {
+            result(newCreateLedgerOp()
+                .withEnsembleSize(1)
+                .withWriteQuorumSize(2)
+                .withAckQuorumSize(1)
+                .withPassword(password)
+                .execute());
+            fail("shoud not be able to create a ledger with such specs");
+        } catch (BKIncorrectParameterException err) {
+        }
+
+        try {
+            result(newCreateLedgerOp()
+                .withEnsembleSize(1)
+                .withWriteQuorumSize(1)
+                .withAckQuorumSize(2)
+                .withPassword(password)
+                .execute());
+            fail("shoud not be able to create a ledger with such specs");
+        } catch (BKIncorrectParameterException err) {
+        }
+
+        try {
+            result(newCreateLedgerOp()
+                .withPassword(null)
+                .execute());
+            fail("shoud not be able to create a ledger with such specs");
+        } catch (BKIncorrectParameterException err) {
+        }
+
+        try {
+            result(newCreateLedgerOp()
+                .withCustomMetadata(null)
+                .withPassword(password)
+                .execute());
+            fail("shoud not be able to create a ledger with such specs");
+        } catch (BKIncorrectParameterException err) {
+        }
+
+        try {
+            ClientConfiguration config = new ClientConfiguration();
+            config.setEnableDigestTypeAutodetection(true);
+            setBookkeeperConfig(config);
+            result(newCreateLedgerOp()
+                .withDigestType(null)
+                .withPassword(password)
+                .execute());
+            fail("shoud not be able to create a ledger with such specs");
+        } catch (BKIncorrectParameterException err) {
+        }
+
+        try {
+            ClientConfiguration config = new ClientConfiguration();
+            config.setEnableDigestTypeAutodetection(false);
+            setBookkeeperConfig(config);
+            result(newCreateLedgerOp()
+                .withDigestType(null)
+                .withPassword(password)
+                .execute());
+            fail("shoud not be able to create a ledger with such specs");
+        } catch (BKIncorrectParameterException err) {
+        }
+
+        closeBookkeeper();
+        try {
+            result(newCreateLedgerOp()
+                .withPassword(password)
+                .execute());
+            fail("shoud not be able to create a ledger, client is closed");
+        } catch (BKClientClosedException err) {
+        }
+    }
+
+    @Test
+    public void testCreateAdvLedger() throws Exception {
```

Comment:
same comment as above.

and ensembleSize, writeQuorumSize, ackQuorumSize, ledgerId, password, customMetadata are used across different test cases. you can define them as class fields.